packages/hdwallet-portis/src/portis.ts

```diff
@@ -0,0 +1,386 @@
+        serialized: result.raw
+    } 
+  }
+
+  public async ethSignMessage (msg: ETHSignMessage): Promise<ETHSignedMessage> {
+
+    const address = await this._ethGetAddress()
+    const result = await this.web3.eth.sign(msg.message, address)
+    return {
+      address,
+      signature: result
+    }
+  }
+
+  public ethGetAccountPaths (msg: ETHGetAccountPath): Array<ETHAccountPath> {
+    return [{
+      addressNList: [ 0x80000000 + 44, 0x80000000 + slip44ByCoin(msg.coin), 0x80000000 + msg.accountIdx, 0, 0 ],
+      hardenedPath: [ 0x80000000 + 44, 0x80000000 + slip44ByCoin(msg.coin), 0x80000000 + msg.accountIdx ],
+      relPath: [ 0, 0 ],
+      description: "Portis"
+    }]
+  }
+
+  public async ethGetAddress (msg: ETHGetAddress): Promise<string> {
+    return this._ethGetAddress()
+  }
+
+  public async getDeviceID(): Promise<string> {
+    return 'portis:' + (await this._ethGetAddress())
+  }
+
+  private async _ethGetAddress(): Promise<string> {
+    return (await this.web3.eth.getAccounts())[0]
+  }
+
+  public async getFirmwareVersion(): Promise<string> {
+    return 'portis'
+  }
+}
+
+export class PortisHDWalletInfo implements HDWalletInfo, ETHWalletInfo {
+  _supportsBTCInfo: boolean = false
+  _supportsETHInfo: boolean = true
+
+  public getVendor (): string {
+    return "Portis"
+  }
+
+  public async ethSupportsNetwork (chainId: number = 1): Promise<boolean> {
+    return chainId === 1
+  }
+
+  public async ethSupportsSecureTransfer (): Promise<boolean> {
+    return false
+  }
+
+  public async ethSupportsNativeShapeShift (): Promise<boolean> {
+    return false
+  }
+
+  public ethGetAccountPaths (msg: ETHGetAccountPath): Array<ETHAccountPath> {
+    return [{
+      addressNList: [ 0x80000000 + 44, 0x80000000 + slip44ByCoin(msg.coin), 0x80000000 + msg.accountIdx, 0, 0 ],
+      hardenedPath: [ 0x80000000 + 44, 0x80000000 + slip44ByCoin(msg.coin), 0x80000000 + msg.accountIdx ],
+      relPath: [ 0, 0 ],
+      description: "Portis"
+    }]
+  }
+
+  public async hasOnDevicePinEntry (): Promise<boolean> {
+    return false
+  }
+
+  public async hasOnDevicePassphrase (): Promise<boolean> {
+    return true
+  }
+
+  public async hasOnDeviceDisplay (): Promise<boolean> {
+    return true
+  }
+
+  public async hasOnDeviceRecovery (): Promise<boolean> {
+    return true
+  }
+
+  public async hasNativeShapeShift (srcCoin: Coin, dstCoin: Coin): Promise<boolean> {
+    // It doesn't... yet?
+    return false
+  }
+
+  public describePath (msg: DescribePath): PathDescription {
+    switch (msg.coin) {
+      case 'Ethereum':
+        return describeETHPath(msg.path)
+      default:
+        throw new Error("Unsupported path")
+      }
+  }
+
+  public ethNextAccountPath (msg: ETHAccountPath): ETHAccountPath | undefined {
+    // Portis only supports one account for eth
+    return undefined
+  }
+}
+
+export function info () {
+  return new PortisHDWalletInfo()
+}
+
+export function create (portis): PortisHDWallet {
```

Comment:
Is it possible to declare the portis param as type Portis? `function create (portis: Portis): PortisHDWallet`